components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -1234,4 +1246,232 @@ public String getOASDefinitionWithTierContentAwareProperty(String oasDefinition,
+     */
+    @Override
+    public Set<URITemplate> injectOtherResourceScopesToDefaultScheme(String resourceConfigsJSON) throws APIManagementException {
+        OpenAPI openAPI = getOpenAPI(resourceConfigsJSON);
+        Set<URITemplate> urlTemplates = new LinkedHashSet<>();
+        Set<Scope> scopes = injectOtherScopesToDefaultScheme(resourceConfigsJSON);
+        List<String> schemes = getOtherSchemes();
+        boolean isDefaultAvailable = isDefaultGiven(resourceConfigsJSON);
+
+        for (String pathKey : openAPI.getPaths().keySet()) {
+            PathItem pathItem = openAPI.getPaths().get(pathKey);
+            for (Map.Entry<PathItem.HttpMethod, Operation> entry : pathItem.readOperationsMap().entrySet()) {
+                Operation operation = entry.getValue();
+                URITemplate template = new URITemplate();
+                if (APIConstants.SUPPORTED_METHODS.contains(entry.getKey().name().toLowerCase())) {
+                    template.setHTTPVerb(entry.getKey().name().toUpperCase());
+                    template.setHttpVerbs(entry.getKey().name().toUpperCase());
+                    template.setUriTemplate(pathKey);
+                    List<String> opScopesofDefault = getScopeOfOperations(OPENAPI_SECURITY_SCHEMA_KEY, operation);
+                    if (opScopesofDefault == null) {
+                        opScopesofDefault = new ArrayList<>();
+                    }
+                    //Handling scopes in resources which do not belong to 'default' scheme
+                    if (!isDefaultAvailable) {
+                        for (int i = 0; i < schemes.size(); i++) {
+                            List<String> opScopesOfOthers = getScopeOfOperations(schemes.get(i), operation);
+                            if (!opScopesOfOthers.isEmpty()) {
+                                if (opScopesOfOthers.size() == 1) {
+                                    String firstScope = opScopesOfOthers.get(0);
+                                    Scope scope = APIUtil.findScopeByKey(scopes, firstScope);
+                                    if (scope == null) {
+                                        throw new APIManagementException("Scope '" + firstScope + "' not found.");
+                                    }
+                                    Scope duplication = scope;
+                                    template.setScope(duplication);
+                                    template.setScopes(duplication);
+                                } else {
+                                    for (String scope : opScopesOfOthers) {
+                                        Scope scopeObj = new Scope();
+                                        scopeObj.setKey(OPENAPI_SECURITY_SCHEMA_KEY);
+                                        scopeObj.setName(OPENAPI_SECURITY_SCHEMA_KEY);
+                                        template.setScopes(scopeObj);
+                                    }
+                                }
```

Comment:
Should we consider the first scope only as APIM GW doesn't support multiple scopes per resource